lib/declarations.d.ts

```diff
@@ -0,0 +1,13 @@
+declare module "resin-discoverable-services" {
+    export interface publishOptions {
+        identifier: string;
+        name: string;
+        host?: string;
+        port: number;
+    }
+
+    export function setRegistryPath(path: string): void;
+    export function enumerateServices(callback?: (error: Error, results: [string]) => void): Promise<[string]>;
```

Comment:
`error` should be nullable as well as the result, right?

And looks like Tim's comment about arrays applies everywhere here